benchmarks/publish-subscribe/src/main.rs

```diff
@@ -42,54 +42,57 @@ fn perform_benchmark<T: Service>(iterations: u64) {
         .history_size(0)
         .subscriber_max_buffer_size(1)
         .enable_safe_overflow(true)
-        .create()
-        .unwrap();
+        .create()?;
 
     let barrier_handle = BarrierHandle::new();
     let barrier = BarrierBuilder::new(3).create(&barrier_handle).unwrap();
 
-    std::thread::scope(|s| {
-        let t1 = s.spawn(|| {
-            let sender_a2b = service_a2b.publisher().create().unwrap();
-            let receiver_b2a = service_b2a.subscriber().create().unwrap();
+    let t1 = ThreadBuilder::new().affinity(0).priority(255).spawn(|| {
+        let sender_a2b = service_a2b.publisher_builder().create().unwrap();
+        let receiver_b2a = service_b2a.subscriber_builder().create().unwrap();
 
-            barrier.wait();
+        barrier.wait();
 
-            for i in 0..iterations {
-                while sender_a2b.send_copy(i).expect("failed to send") == 0 {}
+        let mut sample = sender_a2b.loan().unwrap();
 
-                while receiver_b2a.receive().unwrap().is_none() {}
-            }
-        });
+        for _ in 0..iterations {
+            sample.send().unwrap();
+            sample = sender_a2b.loan().unwrap();
+            while receiver_b2a.receive().unwrap().is_none() {}
+        }
+    });
 
-        let t2 = s.spawn(|| {
-            let sender_b2a = service_b2a.publisher().create().unwrap();
-            let receiver_a2b = service_a2b.subscriber().create().unwrap();
+    let t2 = ThreadBuilder::new().affinity(1).priority(255).spawn(|| {
+        let sender_b2a = service_b2a.publisher_builder().create().unwrap();
+        let receiver_a2b = service_a2b.subscriber_builder().create().unwrap();
 
-            barrier.wait();
+        barrier.wait();
 
-            for i in 0..iterations {
-                while receiver_a2b.receive().unwrap().is_none() {}
+        for _ in 0..iterations {
+            let sample = sender_b2a.loan().unwrap();
+            while receiver_a2b.receive().unwrap().is_none() {}
 
-                while sender_b2a.send_copy(i).expect("failed to send") == 0 {}
-            }
-        });
+            sample.send().unwrap();
+        }
+    });
 
-        std::thread::sleep(std::time::Duration::from_millis(100));
-        let start = Time::now().expect("failed to acquire time");
-        barrier.wait();
+    std::thread::sleep(std::time::Duration::from_millis(100));
+    let start = Time::now().expect("failed to acquire time");
+    barrier.wait();
```

Comment:
Same here ... or do I overlook something?
```suggestion
    barrier.wait();
    let start = Time::now().expect("failed to acquire time");
```

Reply:
See: https://github.com/eclipse-iceoryx/iceoryx2/pull/247#discussion_r1646715246